src/javascript/app_2/Modules/Trading/Containers/trade.jsx

```diff
@@ -23,77 +23,83 @@ class Trade extends React.Component {
         const form_wrapper_class = this.props.is_mobile ? 'mobile-wrapper' : 'sidebar-container desktop-only';
 
         return (
-            <div id='trade_container' className='trade-container'>
-                <div className='chart-container notice-msg'>
-                    { this.props.symbol &&
-                        <SmartChart
-                            chart_id={this.props.chart_id}
-                            InfoBox={<InfoBox is_trade_page />}
-                            onSymbolChange={this.props.onSymbolChange}
-                            symbol={this.props.symbol}
-                            chart_type={this.props.chart_type}
-                            granularity={this.props.granularity}
-                            updateChartType={this.props.updateChartType}
-                            updateGranularity={this.props.updateGranularity}
-                        />
-                    }
-                    <Test />
-                </div>
-                <div
-                    className={form_wrapper_class}
-                >
-                    <FormLayout
-                        is_mobile={this.props.is_mobile}
-                        is_contract_visible={!!contract_id}
-                        is_trade_enabled={this.props.is_trade_enabled}
-                    />
-                    <CSSTransition
-                        in={!!contract_id}
-                        timeout={400}
-                        classNames='contract-wrapper'
-                        unmountOnExit
-                    >
-                        <div className='contract-wrapper'>
-                            <ContractDetails
-                                contract_id={contract_id}
-                                onClickNewTrade={this.props.onClickNewTrade}
+            <React.Fragment>
+                { this.props.is_trade_component_mounted &&
```

Comment:
@kellybinary it causes the `Trade.render` invoked two times, one time when it is mounting and one more time when it is mounted completely.

IMHO, it's better to add a route for the error page.